src/System.CodeDom/tests/System/CodeDom/Compiler/CodeCompilerTests.cs

```diff
@@ -0,0 +1,669 @@
+        }
+
+        [Fact]
+        public void FromFile_NullFileName_ThrowsArgumentNullException()
+        {
+            var compiler = new Compiler();
+            AssertExtensions.Throws<ArgumentNullException>("fileName", () => compiler.FromFileEntryPoint(new CompilerParameters(), null));
+        }
+
+        [Fact]
+        public void FromFile_EmptyFileName_ThrowsArgumentException()
+        {
+            var compiler = new Compiler();
+            AssertExtensions.Throws<ArgumentException>("path", () => compiler.FromFileEntryPoint(new CompilerParameters(), ""));
+        }
+
+        [Fact]
+        public void FromFile_NoSuchFile_ThrowsFileNotFoundException()
+        {
+            var compiler = new Compiler();
+            Assert.Throws<FileNotFoundException>(() => compiler.FromFileEntryPoint(new CompilerParameters(), "noSuchFile"));
+        }
+
+        [Fact]
+        public void CompileAssemblyFromFileBatch_ValidFileNames_ReturnsExpected()
+        {
+            using (var file = new TempFile(Path.GetTempFileName(), 0))
+            {
+                ICodeCompiler compiler = new StubCompiler();
+                Assert.Null(compiler.CompileAssemblyFromFileBatch(new CompilerParameters(), new string[] { file.Path }));
+            }
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void CompileAssemblyFromFileBatch_ValidFileNames_ThrowsPlatformNotSupportedException()
+        {
+            using (var file = new TempFile(Path.GetTempFileName(), 0))
+            {
+                ICodeCompiler compiler = new Compiler();
+                Assert.Throws<PlatformNotSupportedException>(() => compiler.CompileAssemblyFromFileBatch(new CompilerParameters(), new string[] { file.Path }));
+            }
+        }
+
+        [Fact]
+        public void CompileAssemblyFromFileBatch_NullOptions_ThrowsArgumentNullException()
+        {
+            ICodeCompiler compiler = new Compiler();
+            AssertExtensions.Throws<ArgumentNullException>("options", () => compiler.CompileAssemblyFromFileBatch(null, new string[0]));
+        }
+
+        [Fact]
+        public void CompileAssemblyFromFileBatch_NullFileNames_ThrowsArgumentNullException()
+        {
+            ICodeCompiler compiler = new Compiler();
+            AssertExtensions.Throws<ArgumentNullException>("fileNames", () => compiler.CompileAssemblyFromFileBatch(new CompilerParameters(), null));
+        }
+
+        [Fact]
+        public void CompileAssemblyFromFileBatch_NullFileNameInFileNames_ThrowsArgumentNullException()
+        {
+            ICodeCompiler compiler = new Compiler();
+            AssertExtensions.Throws<ArgumentNullException>("path", () => compiler.CompileAssemblyFromFileBatch(new CompilerParameters(), new string[] { null }));
+        }
+
+        [Fact]
+        public void CompileAssemblyFromFileBatch_EmptyFileNameInFileNames_ThrowsArgumentException()
```

Comment:
I expect this is going to fail when run on netfx... probably a few of the other tests as well.

Reply:
Fixed